feature/policy_forwarding/otg_tests/mpls_gre_udp_decap_test/mpls_gre_udp_decap_test.go

```diff
@@ -0,0 +1,582 @@
+	if deviations.ExplicitInterfaceInDefaultVRF(dut) {
+		fptest.AssignToNetworkInstance(t, dut, p.Name(), deviations.DefaultNetworkInstance(dut), 0)
+		t.Logf("DUT %s %s %s requires explicit interface in default VRF deviation ", dut.Vendor(), dut.Model(), dut.Version())
+	}
+
+	if deviations.ExplicitPortSpeed(dut) {
+		fptest.SetPortSpeed(t, p)
+	}
+}
+
+func configureFlow(t *testing.T, config *gosnappi.Config, tc testCase, fc flowConfig, flowName string) {
+	(*config).Flows().Clear()
+	flow := (*config).Flows().Add().SetName(flowName)
+	flow.Metrics().SetEnable(true)
+	flow.TxRx().Device().SetTxNames([]string{fmt.Sprintf("%s.%s", otgPort1.Name, fc.outerIPType)}).SetRxNames([]string{fmt.Sprintf("%s.%s", otgPort2.Name, fc.outerIPType)})
+	flow.Size().SetFixed(trafficFrameSize)
+	flow.Rate().SetPps(trafficRatePps)
+	flow.Duration().SetFixedPackets(gosnappi.NewFlowFixedPackets().SetPackets(noOfPackets))
+
+	eth := flow.Packet().Add().Ethernet()
+	eth.Src().SetValue(otgPort1.MAC)
+
+	switch fc.innerIPType {
+	case ipv4:
+		innerIPv4 := flow.Packet().Add().Ipv4()
+		innerIPv4.Src().SetValue(fc.innerIPSrc)
+		innerIPv4.Dst().SetValue(fc.innerIPDest)
+	case ipv6:
+		innerIPv6 := flow.Packet().Add().Ipv6()
+		innerIPv6.Src().SetValue(fc.innerIPSrc)
+		innerIPv6.Dst().SetValue(fc.innerIPDest)
+	}
+
+	mpls := flow.Packet().Add().Mpls()
+	mpls.Label().SetValue(mplsLabel)
+
+	switch tc.encapType {
+	case gre:
+		flow.Packet().Add().Gre()
+	case gue:
+		udp := flow.Packet().Add().Udp()
+		udp.DstPort().SetValue(outerDstUDPPort)
+	default:
+		t.Errorf("Invalid encap type %s", tc.encapType)
+	}
+
+	switch fc.outerIPType {
+	case ipv4:
+		outerIPv4 := flow.Packet().Add().Ipv4()
+		outerIPv4.Src().SetValue(fc.outerIPSrc)
+		outerIPv4.Dst().SetValue(fc.outerIPDest)
+		outerIPv4.Priority().Dscp().Phb().SetValue(outerDSCP)
+		outerIPv4.TimeToLive().SetValue(outerIPTTL)
+	case ipv6:
+		outerIPv6 := flow.Packet().Add().Ipv6()
+		outerIPv6.Src().SetValue(fc.outerIPSrc)
+		outerIPv6.Dst().SetValue(fc.outerIPDest)
+		outerIPv6.TrafficClass().SetValue(outerDSCP)
+		outerIPv6.HopLimit().SetValue(outerIPTTL)
+	}
+}
+
+func verifyReceivedInnerPacket(t *testing.T, captureFilename string, tc testCase, fc flowConfig) error {
+	if captureFilename == "" {
+		return fmt.Errorf("no capture file provided for inner packet verification for testcase %s", tc.name)
+	}
+
+	handle, err := pcap.OpenOffline(captureFilename)
+	if err != nil {
+		return fmt.Errorf("failed to open pcap file: %v", err)
+	}
+	defer handle.Close()
+
+	var baseIPLayer, icmpLayer gopacket.LayerType
+	switch fc.innerIPType {
+	case ipv4:
+		baseIPLayer = layers.LayerTypeIPv4
+		icmpLayer = layers.LayerTypeICMPv4
+	case ipv6:
+		baseIPLayer = layers.LayerTypeIPv6
+		icmpLayer = layers.LayerTypeICMPv6
+	default:
+		return fmt.Errorf("unknown IP type %s in testcase %s", fc.innerIPType, tc.name)
+	}
+
+	srcIPAddr := net.ParseIP(fc.innerIPSrc)
+	destIPAddr := net.ParseIP(fc.innerIPDest)
+
+	packetCount := 0
```

Comment:
https://google.github.io/styleguide/go/best-practices#declaring-variables-with-zero-values
```suggestion
	var packetCount int
```